test/Autofac.Test/Features/LightweightAdapters/LightweightAdapterRegistrationExtensionsTests.cs

```diff
@@ -81,6 +81,34 @@ public void AdaptedMetadataIsPassed()
             }
         }
 
+        public class DecoratingServiceThatHasDefaultImplementation
+        {
+            readonly IContainer _container;
+
+            public DecoratingServiceThatHasDefaultImplementation()
+            {
+                const string from = "from";
+                var builder = new ContainerBuilder();
+
+                // Implementer1 is the default implementation for IService
+                builder.RegisterType<Implementer1>().As<IService>();
```

Comment:
This is two different services; does it still fail if it's one service?

`builder.RegisterType<Implementer1>().As<IService>().Named<IService>(from);`

If it does fail, can we update it to do that? Or potentially create a second test that uses a single service?